packages/rum_sdk/lib/src/transport/batch_transport.dart

```diff
@@ -26,34 +24,31 @@ class BatchTransport {
 
   Future<void> addEvent(Event event) async {
     payload.events.add(event);
-    items++;
     checkPayloadItemLimit();
   }
 
   Future<void> addMeasurement(Measurement measurement) async {
     payload.measurements.add(measurement);
-    items++;
     checkPayloadItemLimit();
   }
 
   Future<void> addLog(RumLog rumLog) async {
     payload.logs.add(rumLog);
-    items++;
     checkPayloadItemLimit();
   }
 
   Future<void> addExceptions(RumException exception) async {
     payload.exceptions.add(exception);
-    items++;
     checkPayloadItemLimit();
   }
 
   void updatePayloadMeta(Meta meta) {
-    flush();
+    flush(payload.toJson());
+    resetPayload();
     payload.meta = meta;
   }
 
-  Future<void> flush() async {
+  Future<void> flush(Map<String,dynamic> payload)  async {
```

Comment:
duplicate payload gets sent if multiple flush calls were made at once since all flush calls were  using the same payload object